examples/live/create_drm_live_encoding.go

```diff
@@ -0,0 +1,521 @@
+		StreamID:    videoStream1080pResp.Data.Result.ID,
+		MuxingID:    videoTsMuxing1080Resp.Data.Result.ID,
+	}
+	video1080pStreamInfoResponse, err := hlsService.AddStreamInfo(*hlsManifestResp.Data.Result.ID, video1080pStreamInfo)
+	errorHandler(video1080pStreamInfoResponse.Status, err)
+
+	video720pStreamInfo := &models.StreamInfo{
+		Audio:       stringToPtr("audio_group"),
+		SegmentPath: stringToPtr("../video/720p"),
+		URI:         stringToPtr("video_lo.m3u8"),
+		EncodingID:  encodingResp.Data.Result.ID,
+		StreamID:    videoStream720pResp.Data.Result.ID,
+		MuxingID:    videoTsMuxing720Resp.Data.Result.ID,
+	}
+	video720pStreamInfoResponse, err := hlsService.AddStreamInfo(*hlsManifestResp.Data.Result.ID, video720pStreamInfo)
+	errorHandler(video720pStreamInfoResponse.Status, err)
+
+	liveHlsManifest := models.LiveHLSManifest{
+		ManifestID:     hlsManifestResp.Data.Result.ID,
+	}
+
+	fmt.Println("Successfully created Live HLS Manifest!")
+
+	/*
+	Live Stream
+	 */
+	fmt.Println("Waiting for live encoding to be ready...")
+	liveStreamConfig := &models.LiveStreamConfiguration{
+		StreamKey:     stringToPtr("bitmovin"),
+		DashManifests: []models.LiveDashManifest{liveDashManifest},
+		HLSManifests: []models.LiveHLSManifest{liveHlsManifest},
+	}
+
+	startResp, err := encodingS.StartLive(encodingID, liveStreamConfig)
+	errorHandler(startResp.Status, err)
+
+	for numRetries := 0; numRetries < MaxRetries; numRetries++ {
+		time.Sleep(10 * time.Second)
+		statusResp, err := encodingS.RetrieveLiveStatus(encodingID)
+		if err != nil {
+			if err.Error() != "ERROR 2023: Live encoding details not available!" {
+				fmt.Println("Error in starting live encoding")
+				fmt.Println(err)
+				return
+			}
+			fmt.Println("Encoding details not ready yet.")
+			continue
+		}
+		if statusResp != nil {
+			if statusResp.Data.Result.EncoderIP == nil {
+				fmt.Println("Encoder IP detail empty, encoding failed")
+				return
+			}
+			if statusResp.Data.Result.StreamKey == nil {
+				fmt.Println("Stream Key detail empty, encoding failed")
+				return
+			}
+			fmt.Println("---------------")
+			fmt.Println("Live Stream set up successfully:")
+			fmt.Printf("Encoding ID ... %v \n", encodingID)
+			fmt.Printf("Encoder IP .... %v \n", *statusResp.Data.Result.EncoderIP)
+			fmt.Printf("Stream Key .... %v \n", *statusResp.Data.Result.StreamKey)
+			fmt.Printf("Stream URL: ... rtmp://%v/live \n", *statusResp.Data.Result.EncoderIP)
+			fmt.Println("---------------")
+			return
+		}
+	}
+	fmt.Println("Maximum number of retries reached.")
+}
+
+func errorHandler(responseStatus bitmovintypes.ResponseStatus, err error) {
+	if err != nil {
+		fmt.Println("go error")
+		fmt.Println(err)
+	} else if responseStatus == "ERROR" {
+		fmt.Println("api error")
+	}
+}
+
+func stringToPtr(s string) *string {
+	return &s
+}
+
+func intToPtr(i int64) *int64 {
+	return &i
+}
+
+func boolToPtr(b bool) *bool {
+	return &b
+}
+
+func floatToPtr(f float64) *float64 {
+	return &f
+}
```

Comment:
We should put these in a separate helper function library at some point, but let's not worry about it now.